src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+using Xunit;
+
+namespace Microsoft.PowerFx.Tests
+{
+    // Test marshaling between C# objectrs and Power Fx values. 
+    public class MarshalTests
+    {
+        // Do a trivial marshal.
+        [Fact]
+        public void Primitive()
+        {
+            var cache = new TypeMarshallerCache();
+            var tm = cache.GetMarshaler(typeof(int));
+
+            var value = tm.Marshal(5);
+
+            Assert.Equal(5.0, ((NumberValue)value).Value);
+        }
+
+        [Fact]
```

Comment:
What would be the expected result of marshalling an object that we don't support in Power Fx, such as `new int[] { 1, 2, 3 }`? Do we wrap the objects in an object with a Value property (which is what is done in custom connectors that need to talk to APIs that have arrays of primitive values)? Fail?